specifications/trusted_computing_base/key_manager/README.md

```diff
@@ -33,73 +31,127 @@ At a high-level, the KM is designed to be a stand-alone service that drives the
-9. Upon a key export request, the SSB will verify the requester has appropriate permissions to retrieve the requested key, and if so, return the key as required. In this case, it will return **consensus_private<sup>{i+1}</sup>** to LSR. This concludes the consensus key rotation.
+7. At some later point in time, the signed rotation transaction is committed to the blockchain, meaning that when the next consensus reconfiguration event occurs (i.e., the next set of VNs is to selected to perform consensus, see [here](https://github.com/libra/libra/tree/master/specifications/consensus)), the consensus key for the VN will be updated to **consensus_public<sup>{i+1}</sup>**.
+8. Once all 6 steps above have been completed, the consensus key for the VN has successfully been rotated both inside the SS, and on the blockchain. However, due to performance reasons, it's possible that other components within the VN may have previously exported the consensus private key and thus hold temporary copies of the old key locally. For example, LSR is one such component: to improve the throughput and latency of the consensus algorithm, LSR exports the consensus private key upon initialization, and maintains a copy of the private key in local memory. This avoids having to contact the SS whenever it wishes to sign a quorum certificate. In order to prevent this key from becoming stale, LSR monitors the blockchain for reconfiguration events, and after each reconfiguration event, LSR will verify if the consensus key for this VN has been rotated. If so, LSR will query the SS directly to export the consensus private key corresponding to the consensus public key registered in the validator config on-chain (i.e., fetch **consensus_private<sup>{i+1}</sup>** for **consensus_public<sup>{i+1}</sup>**).
+9. Upon a key export request, the SS will verify the requester has appropriate permissions to retrieve the requested key, and if so, return the key as required. In this case, it will return **consensus_private<sup>{i+1}</sup>** to LSR. This concludes the consensus key rotation.
 
 ### Consensus Key Rotation Failures
```

Comment:
might want to have an assumptions paragraph

* the key has been properly instantiated
* only one instance of KM is managing the key
* the json rpc service is honest

etc